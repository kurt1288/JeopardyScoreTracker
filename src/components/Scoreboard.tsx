import { useState } from "react";
import { useSelector } from "react-redux";
import { RootState } from "../store";

const Scoreboard = () => {
    const { players } = useSelector((state: RootState) => state);
    const [collapsed, setCollapsed] = useState(true);
    
    return (
        <div id="scoreboard" className={collapsed ? 'collapsed' : ''} onClick={() => setCollapsed(!collapsed)} role="button">
            <div id="scores">
                {players.map((player, index) => (
                    <p key={index}>{ player.name }: { player.score }</p>
                ))}
            </div>
            <div id="points">
                {players.map(player => (
                    <div>
                        { player.points.map((obj, index) => (
                            <div>
                                <p>Round {index + 1}</p>
                                <ul>
                                    { Object.entries(obj).map(x => (
                                        <li>{x[0]}: {x[1]}</li>
                                    )) }
                                </ul>
                            </div>
                        )) }
                    </div>
                ))}
            </div>
        </div>
    )
}

export default Scoreboard;