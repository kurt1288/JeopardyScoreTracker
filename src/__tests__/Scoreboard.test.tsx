import * as redux from 'react-redux';
import Scoreboard from '../components/Scoreboard';
import { fireEvent, render } from '@testing-library/react';

const useSelectorMock = jest.spyOn(redux, 'useSelector');
const useDispatchMock = jest.spyOn(redux, 'useDispatch');

const players = [{name:'Jack',points:[{'200':0,'400':0,'600':0,'800':0,'1000':0},{'400':0,'800':0,'1200':0,'1600':0,'2000':0},{'5000':0}],score:0},{name:'Jill',points:[{'200':0,'400':0,'600':0,'800':0,'1000':0},{'400':0,'800':0,'1200':0,'1600':0,'2000':0},{'5000':0}],score:0}];

beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
});

test("shows score", () => {
    useSelectorMock.mockReturnValue({players: players});

    const screen = render(<Scoreboard />);

    expect(screen).toMatchSnapshot();
});

test("shows full scoreboard", () => {
    useSelectorMock.mockReturnValue({players: players});

    const screen = render(<Scoreboard />);

    fireEvent.click(screen.getByRole("button"));

    expect(screen.getByRole("button")).toBeInTheDocument();
});