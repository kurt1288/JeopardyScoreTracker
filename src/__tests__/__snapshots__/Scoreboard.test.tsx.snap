// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows score 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="collapsed"
        id="scoreboard"
        role="button"
      >
        <div
          id="scores"
        >
          <p>
            Jack
            : 
            0
          </p>
          <p>
            Jill
            : 
            0
          </p>
        </div>
        <div
          id="points"
        >
          <div>
            <div>
              <p>
                Round 
                1
              </p>
              <ul>
                <li>
                  200
                  : 
                  0
                </li>
                <li>
                  400
                  : 
                  0
                </li>
                <li>
                  600
                  : 
                  0
                </li>
                <li>
                  800
                  : 
                  0
                </li>
                <li>
                  1000
                  : 
                  0
                </li>
              </ul>
            </div>
            <div>
              <p>
                Round 
                2
              </p>
              <ul>
                <li>
                  400
                  : 
                  0
                </li>
                <li>
                  800
                  : 
                  0
                </li>
                <li>
                  1200
                  : 
                  0
                </li>
                <li>
                  1600
                  : 
                  0
                </li>
                <li>
                  2000
                  : 
                  0
                </li>
              </ul>
            </div>
            <div>
              <p>
                Round 
                3
              </p>
              <ul>
                <li>
                  5000
                  : 
                  0
                </li>
              </ul>
            </div>
          </div>
          <div>
            <div>
              <p>
                Round 
                1
              </p>
              <ul>
                <li>
                  200
                  : 
                  0
                </li>
                <li>
                  400
                  : 
                  0
                </li>
                <li>
                  600
                  : 
                  0
                </li>
                <li>
                  800
                  : 
                  0
                </li>
                <li>
                  1000
                  : 
                  0
                </li>
              </ul>
            </div>
            <div>
              <p>
                Round 
                2
              </p>
              <ul>
                <li>
                  400
                  : 
                  0
                </li>
                <li>
                  800
                  : 
                  0
                </li>
                <li>
                  1200
                  : 
                  0
                </li>
                <li>
                  1600
                  : 
                  0
                </li>
                <li>
                  2000
                  : 
                  0
                </li>
              </ul>
            </div>
            <div>
              <p>
                Round 
                3
              </p>
              <ul>
                <li>
                  5000
                  : 
                  0
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="collapsed"
      id="scoreboard"
      role="button"
    >
      <div
        id="scores"
      >
        <p>
          Jack
          : 
          0
        </p>
        <p>
          Jill
          : 
          0
        </p>
      </div>
      <div
        id="points"
      >
        <div>
          <div>
            <p>
              Round 
              1
            </p>
            <ul>
              <li>
                200
                : 
                0
              </li>
              <li>
                400
                : 
                0
              </li>
              <li>
                600
                : 
                0
              </li>
              <li>
                800
                : 
                0
              </li>
              <li>
                1000
                : 
                0
              </li>
            </ul>
          </div>
          <div>
            <p>
              Round 
              2
            </p>
            <ul>
              <li>
                400
                : 
                0
              </li>
              <li>
                800
                : 
                0
              </li>
              <li>
                1200
                : 
                0
              </li>
              <li>
                1600
                : 
                0
              </li>
              <li>
                2000
                : 
                0
              </li>
            </ul>
          </div>
          <div>
            <p>
              Round 
              3
            </p>
            <ul>
              <li>
                5000
                : 
                0
              </li>
            </ul>
          </div>
        </div>
        <div>
          <div>
            <p>
              Round 
              1
            </p>
            <ul>
              <li>
                200
                : 
                0
              </li>
              <li>
                400
                : 
                0
              </li>
              <li>
                600
                : 
                0
              </li>
              <li>
                800
                : 
                0
              </li>
              <li>
                1000
                : 
                0
              </li>
            </ul>
          </div>
          <div>
            <p>
              Round 
              2
            </p>
            <ul>
              <li>
                400
                : 
                0
              </li>
              <li>
                800
                : 
                0
              </li>
              <li>
                1200
                : 
                0
              </li>
              <li>
                1600
                : 
                0
              </li>
              <li>
                2000
                : 
                0
              </li>
            </ul>
          </div>
          <div>
            <p>
              Round 
              3
            </p>
            <ul>
              <li>
                5000
                : 
                0
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
